great_expectations/core/expectation_suite.py

```diff
@@ -122,15 +128,34 @@ def __init__(  # noqa: PLR0913
 
         if expectations is None:
             expectations = []
-        expectation_configurations = [
-            ExpectationConfiguration(**expectation)
-            if isinstance(expectation, dict)
-            else expectation
-            for expectation in expectations
-        ]
-        self.expectations: list[Expectation] = [
-            self._build_expectation(config) for config in expectation_configurations
-        ]
+
+        if all(isinstance(exp, Expectation) for exp in expectations):
+            for expectation in expectations:
+                cast(Expectation, expectation)
+                if expectation.id:
+                    raise ValueError(
+                        "Expectations in parameter `expectations` must not belong to another ExpectationSuite. "
+                        "Instead, please use copies of Expectations, by calling `copy.copy(expectation)`."
+                    )
+                expectation.register_save_callback(save_callback=self._save_expectation)
+            self.expectations = expectations
```

Comment:
Alright, final comment. Sorry for so much feedback about something so minor! If mypy is sad about this, then a simple solution might be to replace all of this logic in the constructor with `self.expectations = ExpectationSuite._ensure_expectation([exp for exp in expectations or []])`. Implementing `_ensure_expectation` is left as an exercise to the reader.